src/plays/password-generator/PasswordGenerator.jsx

```diff
@@ -58,89 +61,107 @@ function PasswordGenerator(props) {
 
   // Generate a random password
   const generatePassword = () => {
-    setError(false);
-    const concated = arrangeData();
-    let finalPassword = "";
-    for (let i = 0; i < passwordConfig.length; i++) {
-      finalPassword += concated[randomNumberGenerator(concated.length)];
-    }
-    return finalPassword;
+      setError(false);
+      const concated = arrangeData();
+      let finalPassword = "";
+      for (let i = 0; i < passwordConfig.length; i++) {
+        finalPassword += concated[randomNumberGenerator(concated.length)];
+      }
+      return finalPassword;
   };
 
   // generate password button click handler
   const generateHander = () => {
     const finalPassword = generatePassword();
     setPassword({ status: false, password: finalPassword });
+    passwordField.current.value = finalPassword;
```

Comment:
we already have the password in the password state so why we need to manage ref?